### DISCUSSION TITLE
Schema Files with Java Classes

### DISCUSSION THREAD

**Discussion by LexanRed** [Netflix/dgs-framework#627] [Q&A]
Hi there, 

is it possible to place schema files in the same package as Java Classes? 
In my opinion it is beneficial to place the schema file directly next to a model / entity class. 

We have lots of entities which are mapped to the database and most of them als have a .graphql file in the schema folder. 
We create a file for every package and separate our schemas there. 

I tried to set a path with the following option in the application.properties file:
```dgs.graphql.schema-locations="**/*.graphql*"```
The default is the following
```dgs.graphql.schema-locations="classpath*:schema/**/*.graphql*"```
It did not work as expected. 

If it is not possible, why isn't it? Is it discouraged by any reason?

Any help is appreciated :)

**Comment by paulbakker** [comment]
The framework searches for schema files in `META-INF/schema` In JAR files. We use this for creating libraries containing common types that are used by multiple services.

**Comment by LexanRed** [comment]
Ok. Got that. 

Since I can define the path in application.properties where to look for my own schemas, is it possible to adapt the path to achieve the described situation?

<img width="312" alt="Bildschirmfoto 2021-09-25 um 14 15 44" src="https://user-images.githubusercontent.com/57357219/134771187-d49da19a-dcd2-46ee-9267-01035f285ac1.png">

Instead of putting all schemas in a subfolder of resources?

**Reply by paulbakker** [reply]
It *always* checks for `META-INF/schema` files, so you don't need extra configuration for that.

**Reply by LexanRed** [reply]
I understood that, it is just not how we would like to organise our code base.

So is it possible, to tell DGS to look for schemas next to classes or is it necessary to put the schemas into the resources folder?

**Reply by paulbakker** [reply]
I missed the part about having the schema files next to the java classes. That not currently possible (and probably won't be), because it would require to scan jar files, which is really expensive.
